### PR TITLE
[JENKINS-60608] - Rename the `jenkins-slave.exe` binary to `jenkins-agent.exe` for newly created agents

### PR DESCRIPTION
**Pull request by jsoref**
See jenkinsci/jenkins@434135e

### CODE REVIEW

**Review by jvz — COMMENTED** [review]
I'm generally onboard with this. I'm not sure about any dependencies around releasing things for the renaming, though.

**Comment by oleg-nenashev** [comment]
It would be nice to add a placeholder task in https://issues.jenkins-ci.org/browse/JENKINS-42816 so that we can track overall progress for "slave" leftovers. @jsoref if you see other changes missing in plugins, tickets will be also appreciated (especially newbie-friendly ones so that we can leave them to newcomers)

File: src/main/resources/hudson/os/windows/ManagedWindowsServiceLauncher/config.properties [hunk 1]
```diff
@@ -1,3 +1,3 @@
 blurb=\
-    This launch method relies on DCOM and is often associated with <a href="https://wiki.jenkins-ci.org/display/JENKINS/Windows+slaves+fail+to+start+via+DCOM" target="_blank">subtle problems</a>. \
+    This launch method relies on DCOM and is often associated with <a href="https://wiki.jenkins-ci.org/display/JENKINS/Windows+agents+fail+to+start+via+DCOM" target="_blank">subtle problems</a>. \
```
Comment (oleg-nenashev) [inline src/main/resources/hudson/os/windows/ManagedWindowsServiceLauncher/config.properties:2]: Wiki page has been already renamed, so we want to land it ASAP I'd guess

File: src/main/java/hudson/os/windows/ManagedWindowsServiceLauncher.java [hunk 2]
```diff
@@ -509,12 +509,12 @@ public void afterDisconnect(SlaveComputer computer, TaskListener listener) {
         }
     }
 
-    String generateServiceId(String slaveRoot) throws IOException {
-        return "jenkinsslave-" + slaveRoot.replace(':', '_').replace('\\', '_').replace('/', '_');
+    String generateServiceId(String agentRoot) throws IOException {
```
Comment (oleg-nenashev) [inline src/main/java/hudson/os/windows/ManagedWindowsServiceLauncher.java:517]: IIUC it will break existing services

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Thanks @jsoref ! I think it is important to land this change, but it should ensure that previously created agent services with "slave-..." continue working as is. Otherwise we risk serious collisions on the agent side since conflicting services might be created

File: src/main/java/hudson/os/windows/ManagedWindowsServiceLauncher.java (hunk 2 above)
Comment (jsoref) [reply src/main/java/hudson/os/windows/ManagedWindowsServiceLauncher.java:517]: Hmm... Looks like it needs to try both?

Comment (jsoref) [reply src/main/java/hudson/os/windows/ManagedWindowsServiceLauncher.java:517]: I think I've addressed this...

**Comment by oleg-nenashev** [comment]
missed the notification

File: src/main/java/hudson/os/windows/ManagedWindowsServiceLauncher.java (hunk 2 above)
Comment (oleg-nenashev) [reply src/main/java/hudson/os/windows/ManagedWindowsServiceLauncher.java:517]: Looks so

**Comment by oleg-nenashev** [comment]
Created https://issues.jenkins-ci.org/browse/JENKINS-60608

File: src/main/resources/hudson/os/windows/ManagedWindowsServiceLauncher/config.properties (hunk 1 above)
Comment (oleg-nenashev) [reply src/main/resources/hudson/os/windows/ManagedWindowsServiceLauncher/config.properties:2]: I will fix it in a follow-up. Now it is https://github.com/jenkinsci/windows-slaves-plugin/blob/master/docs/troubleshooting.adoc

**Review by oleg-nenashev — APPROVED** [review]
I am approving it for the new release

**Comment by oleg-nenashev** [comment]
Released it in https://github.com/jenkinsci/windows-slaves-plugin/releases/tag/windows-slaves-1.6

**Pull request merged**